src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs

```diff
@@ -80,88 +77,121 @@ public sealed class SqlServerFhirModel : IStartable
+            ThrowIfNotInitialized();
             return _resourceTypeToId[resourceTypeName];
         }
 
         public bool TryGetResourceTypeId(string resourceTypeName, out short id)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _resourceTypeToId.TryGetValue(resourceTypeName, out id);
         }
 
         public string GetResourceTypeName(short resourceTypeId)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _resourceTypeIdToTypeName[resourceTypeId];
         }
 
         public byte GetClaimTypeId(string claimTypeName)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _claimNameToId[claimTypeName];
         }
 
         public short GetSearchParamId(Uri searchParamUri)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _searchParamUriToId[searchParamUri];
         }
 
         public byte GetCompartmentTypeId(string compartmentType)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _compartmentTypeToId[compartmentType];
         }
 
         public bool TryGetSystemId(string system, out int systemId)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _systemToId.TryGetValue(system, out systemId);
         }
 
         public int GetSystemId(string system)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
 
             VLatest.SystemTable systemTable = VLatest.System;
             return GetStringId(_systemToId, system, systemTable, systemTable.SystemId, systemTable.Value);
         }
 
         public int GetQuantityCodeId(string code)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
 
             VLatest.QuantityCodeTable quantityCodeTable = VLatest.QuantityCode;
             return GetStringId(_quantityCodeToId, code, quantityCodeTable, quantityCodeTable.QuantityCodeId, quantityCodeTable.Value);
         }
 
         public bool TryGetQuantityCodeId(string code, out int quantityCodeId)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _quantityCodeToId.TryGetValue(code, out quantityCodeId);
         }
 
-        // TODO: Once this no longer implements IStartable, rename to "EnsureInitialized" (work item 75558).
-        public void Start()
+        public Task EnsureInitialized()
         {
-            _schemaInitializer.Start();
-
-            if (_searchParameterDefinitionManager is IStartable startable)
+            if (_schemaInformation.Current == null)
             {
-                startable.Start();
+                // TODO: What is the correct error to throw here?
+                throw new NullReferenceException("The current schema version should not be null");
```

Comment:
I don't think that `NullReferenceException` is the right exception to use here. `_schemaInformation.Current` would only be `null` if there is a server error. Should we have this check at all?

Reply:
InvalidOperationException would be a better choice. NullReferenceException is not really meant to be thrown by user code. I think it's good to keep the check.

Reply:
Cool thanks, noted for next time. Fixed in c1d9233.